jetstream/analysis.py

```diff
@@ -20,6 +19,37 @@
 from jetstream.config import AnalysisConfiguration
 from jetstream.dryrun import dry_run_query
 from jetstream.statistics import Count, StatisticResult, StatisticResultCollection
+from jetstream.logging import logger
+
+
+class NoSlugException(Exception):
+    """Experiment has no slug."""
+
+    pass
+
+
+class NoEnrollmentPeriodException(Exception):
+    """Experiment has no enrollment period."""
+
+    pass
+
+
+class NoStartDateException(Exception):
+    """Experiment has no start date."""
+
+    pass
+
+
+class EndedException(Exception):
+    """Experiment has already ended."""
+
+    pass
+
+
+class EnrollmentLongerThanAnalysisException(Exception):
+    """Enrollment period is longer than analysis dates"""
+
+    pass
```

Comment:
Not sure if this is necessary. It's useful being able to filter/group the logs based on the exception type.

Reply:
Since we only use each of these once I'd be tempted to combine them into an `ExperimentRejected` type but this also seems cool.

If we have a different type for each error, we could make the types responsible for the string formatting.

Reply:
Added string formatting to the custom exceptions